### PR TITLE
Remove protocols wrapper for QtQuick

### PR DESCRIPTION
**Pull request by zccrs**

### CODE REVIEW

**Comment by wineee** [comment]
```
waylib-0.1.1> [ 97%] Building CXX object examples/tinywl/CMakeFiles/tinywl-qtquick.dir/.rcc/qmlcache/tinywl-qtquick_WindowDecoration_qml.cpp.o
waylib-0.1.1> [ 97%] Building CXX object examples/tinywl/CMakeFiles/tinywl-qtquick.dir/.rcc/qmlcache/tinywl-qtquick_CloseAnimation_qml.cpp.o
waylib-0.1.1> /build/source/examples/tinywl/main.cpp: In lambda function:
waylib-0.1.1> /build/source/examples/tinywl/main.cpp:234:17: error: 'QWGammaControl' has not been declared
waylib-0.1.1>   234 |                 QWGammaControl::from(gamma_control)->sendFailedAndDestroy();
waylib-0.1.1>       |                 ^~~~~~~~~~~~~~
waylib-0.1.1> [ 97%] Built target blur
waylib-0.1.1> [ 98%] Building CXX object examples/tinywl/CMakeFiles/tinywl-qtquick.dir/.rcc/qmlcache/tinywl-qtquick_MiniDock_qml.cpp.o
waylib-0.1.1> [ 98%] Building CXX object examples/tinywl/CMakeFiles/tinywl-qtquick.dir/.rcc/qmlcache/tinywl-qtquick_InputPopupSurface_qml.cpp.o
waylib-0.1.1> [ 98%] Building CXX object examples/tinywl/CMakeFiles/tinywl-qtquick.dir/.qt/rcc/qrc_tinywl-qtquick_raw_qml_0.cpp.o
```

**Comment by zccrs** [comment]
Fix https://github.com/vioken/waylib/issues/385

File: examples/blur/helper.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#include <wglobal.h>
+#include <wqmldynamiccreator.h>
+
+#include <QObject>
+#include <QQmlEngine>
+
+WAYLIB_SERVER_BEGIN_NAMESPACE
+class WServer;
+class WOutputRenderWindow;
+class WQmlCreator;
```
Comment (wineee) [inline examples/blur/helper.h:15]: It's better just include, Q_PROPERTY need complete definition

**Review by wineee — CHANGES_REQUESTED** [review]

File: examples/blur/helper.h (hunk 1 above)
Comment (wineee) [inline examples/blur/helper.h:15]:
```suggestion
```

**Review by wineee — CHANGES_REQUESTED** [review]

File: examples/tinywl/helper.h [hunk 2]
```diff
@@ -4,16 +4,38 @@
...
+class WQmlCreator;
+class WXdgOutputManager;
+class WXdgDecorationManager;
+class WInputMethodHelper;
+class WCursorShapeManagerV1;
+class WOutputManagerV1;
+WAYLIB_SERVER_END_NAMESPACE
+
+QW_BEGIN_NAMESPACE
+class QWCompositor;
```
Comment (wineee) [inline examples/tinywl/helper.h:34]:
```suggestion
```

File: examples/tinywl/helper.h [hunk 3]
```diff
@@ -4,16 +4,38 @@
...
+WAYLIB_SERVER_BEGIN_NAMESPACE
+class WQuickCursor;
+class WOutputRenderWindow;
+class WQmlCreator;
+class WXdgOutputManager;
+class WXdgDecorationManager;
+class WInputMethodHelper;
+class WCursorShapeManagerV1;
+class WOutputManagerV1;
+WAYLIB_SERVER_END_NAMESPACE
```
Comment (wineee) [inline examples/tinywl/helper.h:31]:
```suggestion
WAYLIB_SERVER_BEGIN_NAMESPACE
class WQuickCursor;
class WOutputRenderWindow;
class WXdgOutputManager;
class WInputMethodHelper;
class WCursorShapeManagerV1;
class WOutputManagerV1;
WAYLIB_SERVER_END_NAMESPACE
```

File: examples/outputviewport/helper.h [hunk 4]
```diff
@@ -3,19 +3,56 @@
...
+#include <wglobal.h>
+#include <wqmldynamiccreator.h>
 
 #include <QObject>
 #include <QQmlEngine>
 
+WAYLIB_SERVER_BEGIN_NAMESPACE
+class WServer;
+class WOutputRenderWindow;
+class WQmlCreator;
```
Comment (Groveer) [inline examples/outputviewport/helper.h]:
```suggestion

```

File: examples/tinywl/helper.h [hunk 5]
```diff
@@ -79,18 +127,42 @@ public Q_SLOTS:
...
     OutputInfo* getOutputInfo(WOutput *output);
 
+    WServer *m_server = nullptr;
+    QWRenderer *m_renderer = nullptr;
+    QWAllocator *m_allocator = nullptr;
+    QWCompositor *m_compositor = nullptr;
+    WQuickOutputLayout *m_outputLayout = nullptr;
+    WQuickCursor *m_cursor = nullptr;
+    QPointer<WSeat> m_seat;
+    QPointer<WXWayland> m_xwayland;
```
Comment (wineee) [inline examples/tinywl/helper.h:137]: QPointer or c pointer，should keep same？

File: examples/outputviewport/main.cpp [hunk 6]
```diff
@@ -14,9 +33,99 @@
 #include <QQuickItem>
 #include <QQuickWindow>
 
-WAYLIB_SERVER_USE_NAMESPACE
+extern "C" {
```
Comment (Groveer) [inline examples/outputviewport/main.cpp:35]: 不需要了吧

File: examples/tinywl/helper.h (hunk 5 above)
Comment (zccrs) [reply examples/tinywl/helper.h:137]: No, only use with WSeat and WXWayland.

**Review by Groveer — APPROVED** [review]

File: examples/tinywl/main.cpp [hunk 7]
```diff
@@ -7,14 +7,38 @@
...
+#include <qwallocator.h>
+#include <qwrenderer.h>
+#include <qwcompositor.h>
+#include <qwsubcompositor.h>
+#include <qwxwaylandsurface.h>
+#include <qwlayershellv1.h>
+#include <qwscreencopyv1.h>
+#include <qwfractionalscalemanagerv1.h>
+#include <qwgammacontorlv1.h>
+#include <woutputitem.h>
```
Comment (wineee) [inline examples/tinywl/main.cpp:40]: 头文件顺序

**Review by wineee — APPROVED** [review]

**Pull request merged**